expression/builtin_like_vec.go

```diff
@@ -127,9 +127,14 @@ func (b *builtinRegexpSharedSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 		return err
 	}
 
-	if b.args[1].ConstItem(b.ctx.GetSessionVars().StmtCtx) && !b.isMemorizedRegexpInitialized() {
-		b.initMemoizedRegexp(bufPat, n)
+	memorization := func() {
+		if b.args[1].ConstItem(b.ctx.GetSessionVars().StmtCtx) && !b.isMemorizedRegexpInitialized() {
+			b.initMemoizedRegexp(bufPat, n)
+		}
 	}
+	// Only be executed once to achieve thread-safe
+	b.once.Do(memorization)
```

Comment:
I think there are some problems. Maybe after calling `once.Do(memorization)`, the memorizedRegexp is also not initialized, for example #16405, the input rownum is zero.